lib/msprime.c

```diff
@@ -7018,6 +7018,16 @@ genic_selection_stochastic_forwards(double dt, double freq, double alpha, double
     return freq + (ux * dt) + sign * sqrt(freq * (1.0 - freq) * dt);
 }
 
+static double
+neutral_stochastic_backwards(double dt, double freq, double u)
+{
+    /* returns allele freq of neutral allele drifting conditional
+    on loss (looking backwards in time) following Ewens */
+    double ux = -1.0 * freq;
+    int sign = u < 0.5 ? 1 : -1;
+    return freq + (ux * dt) + sign * sqrt(freq * (1.0 - freq) * dt);
```

Comment:
Can we think of reasonable ways to handle the possibility of numeric errors here?  It seems that there are no restrictions placed on the input values, so one can get negative frequencies out:

```python
In [1]: dt = 1e-8

In [2]: freq = 1e-8

In [3]: ux = -freq

In [4]: sign = -1

In [5]: import math

In [7]: freq + (ux * dt) + sign * math.sqrt(freq * (1.0 - freq) * dt)
Out[7]: -4.999999918875795e-17
```

Perhaps this should return `max(value, 0.0)` ?